crates/aptos/src/node/local_testnet/mod.rs

```diff
@@ -222,10 +229,27 @@ impl CliCommand<()> for RunLocalTestnet {
             setup_logging(Some(Box::new(make_writer)));
         }
 
+        // If the CLI is running inside a container, bind services not running inside a
+        // container to 0.0.0.0 (so they can be accessed from outside the container).
+        // Otherwise bind them to 127.0.0.1. This is necessary because Windows
+        // complains about services binding to 0.0.0.0 sometimes.
+        let running_inside_container = Path::new(".dockerenv").exists();
+        let bind_to = match self.bind_to {
+            Some(bind_to) => bind_to,
+            None => {
+                if running_inside_container {
+                    Ipv4Addr::new(0, 0, 0, 0)
+                } else {
+                    Ipv4Addr::new(127, 0, 0, 1)
+                }
```

Comment:
Did you want to gate the default based on OS, or just stick with this?

Reply:
Detecting OS is a rabbithole I've gone down before and would rather not do again lol, much better this way.